lib/ansible/modules/network/cloudengine/ce_static_route.py

```diff
@@ -0,0 +1,845 @@
+
+        if destvrf is None:
+            dest_vpn_instance = "_public_"
+        else:
+            dest_vpn_instance = destvrf
+        if self.description:
+            description_xml = CE_NC_SET_DESCRIPTION % self.description
+        if self.pref:
+            preference_xml = CE_NC_SET_PREFERENCE % self.pref
+        if self.tag:
+            tag_xml = CE_NC_SET_TAG % self.tag
+
+        if state == "present":
+            configxmlstr = CE_NC_SET_STATIC_ROUTE % (
+                vpn_instance, version, prefix, mask, nhp_interface,
+                dest_vpn_instance, next_hop, description_xml, preference_xml, tag_xml)
+        else:
+            configxmlstr = CE_NC_DELETE_STATIC_ROUTE % (
+                vpn_instance, version, prefix, mask, nhp_interface, dest_vpn_instance, next_hop)
+
+        conf_str = build_config_xml(configxmlstr)
+
+        recv_xml = set_nc_config(self.module, conf_str)
+        self.check_response(recv_xml, "OPERATE_STATIC_ROUTE")
+
+    def get_static_route(self, state):
+        """get ipv4 static route"""
+
+        self.static_routes_info["sroute"] = list()
+
+        if state == 'absent':
+            getxmlstr = CE_NC_GET_STATIC_ROUTE_ABSENT
+        else:
+            getxmlstr = CE_NC_GET_STATIC_ROUTE
+
+        xml_str = get_nc_config(self.module, getxmlstr)
+
+        if 'data/' in xml_str:
+            return
+        xml_str = xml_str.replace('\r', '').replace('\n', '').\
+            replace('xmlns="urn:ietf:params:xml:ns:netconf:base:1.0"', "").\
+            replace('xmlns="http://www.huawei.com/netconf/vrp"', "")
+        root = ElementTree.fromstring(xml_str)
+        static_routes = root.findall(
+            "data/staticrt/staticrtbase/srRoutes/srRoute")
+
+        if static_routes:
+            for static_route in static_routes:
+                static_info = dict()
+                for static_ele in static_route:
+                    if static_ele.tag in ["vrfName", "afType", "topologyName",
+                                          "prefix", "maskLength", "destVrfName",
+                                          "nexthop", "ifName", "preference", "description"]:
+                        static_info[
+                            static_ele.tag] = static_ele.text
+                    if static_ele.tag == "tag":
+                        if static_ele.text is not None:
+                            static_info["tag"] = static_ele.text
+                        else:
+                            static_info["tag"] = "None"
+                self.static_routes_info["sroute"].append(static_info)
+
+    def check_params(self):
+        """check all input params"""
+        # prefix, mask, aftype, next_hop, state, check
+        if not self.prefix or not self.mask or not self.aftype or not self.state:
+            self.module.fail_json(
+                msg='Error: Prefix or mask or address family type or state must be set.')
+        if not self.next_hop and not self.nhp_interface:
```

Comment:
There are some options to `AnsibleModule` which you may be able to use to validate options passed in.
Look at existing modules for examples:
```
mutually_exclusive
required_together
required_one_of
require_if
```